frontend/src/app/signup/page.tsx

```diff
@@ -22,11 +22,15 @@ const FormSchema = z.object({
     .email({ message: "Invalid email." }),
   password: z
     .string()
-    .min(1, { message: "Please enter a password." }),
+    .min(1, { message: "Please enter a password." })
+    .refine((val) => val.length >= 6 && ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9'].includes(val), {
```

Comment:
I think we need `/[0-9]/.test(val)` to check if there is a number. The current check seems to be failing even when there is a number